pkg/generate/app/cmd/newapp_test.go

```diff
@@ -615,6 +651,22 @@ func TestRunAll(t *testing.T) {
 				imageStreams = append(imageStreams, tp)
 			case *deploy.DeploymentConfig:
 				got["deploymentConfig"] = append(got["deploymentConfig"], tp.Name)
+				if podTemplate := tp.Template.ControllerTemplate.Template; podTemplate != nil {
+					for _, volume := range podTemplate.Spec.Volumes {
+						var srcType string
+						if volume.VolumeSource.EmptyDir != nil {
+							srcType = "EmptyDir"
+						} else {
+							srcType = "UNKNOWN"
+						}
+						got["volumes"] = append(got["volumes"], volume.Name+"♥"+srcType)
```

Comment:
@mfojtik @pmorie needed a separator in the string, wanted something cheerful. Since we're living in an utf-8 age now and these are just tests anyway, didn't see any harm in it.